todomvc/src/lib.rs

```diff
@@ -0,0 +1,158 @@
+use anyhow::{anyhow, Result};
+use guest::prelude::*;
+use http::{Request, Response};
+use log::{info, trace, warn};
+use serde::{Deserialize, Serialize};
+use wapc_guest as guest;
+use wasmcloud_actor_core as actor;
+use wasmcloud_actor_http_server as http;
+use wasmcloud_actor_keyvalue as kv;
+use wasmcloud_actor_logging as logging;
+
+#[actor::init]
+pub fn init() {
+    http::Handlers::register_handle_request(request_handler);
+    logging::enable_macros();
+}
+
+#[derive(Serialize, Deserialize)]
+struct InputTodo {
+    title: String,
+    order: Option<i32>,
+}
+#[derive(Serialize, Deserialize)]
+struct UpdateTodo {
+    title: Option<String>,
+    completed: Option<bool>,
+    order: Option<i32>,
+}
+#[derive(Serialize, Deserialize)]
+struct Todo {
+    url: String,
+    title: String,
+    completed: bool,
+    order: i32,
+}
+
+impl Todo {
+    fn new(url: String, title: String, order: i32) -> Self {
+        Self {
+            url,
+            title,
+            completed: false,
+            order,
+        }
+    }
```

Comment:
`fn update(self, update: UpdateTodo) -> Todo {...}`

and `let todo = todo.update(update)` below

Reply:
let's do this